internal/storagenode/logstream/executor.go

```diff
@@ -352,13 +352,17 @@ func (lse *Executor) Report(_ context.Context) (snpb.LogStreamUncommitReport, er
 		return snpb.LogStreamUncommitReport{}, verrors.ErrClosed
 	}
 
-	version, highWatermark, uncommittedLLSNBegin := lse.lsc.reportCommitBase()
+	// The llsnOffset specifies the previous commit context; on the other
+	// hand, the uncommittedLLSNBegin indicates the starting position of
+	// log entries written but not committed.
+	version, highWatermark, llsnOffset := lse.lsc.reportCommitBase()
+	uncommittedLLSNBegin := lse.lsc.uncommittedLLSNBegin.Load()
 	uncommittedLLSNEnd := lse.lsc.uncommittedLLSNEnd.Load()
 	report := snpb.LogStreamUncommitReport{
 		LogStreamID:           lse.lsid,
 		Version:               version,
 		HighWatermark:         highWatermark,
-		UncommittedLLSNOffset: uncommittedLLSNBegin,
+		UncommittedLLSNOffset: llsnOffset,
```

Comment:
Since `llsnOffset` and `uncommittedLLSNBegin` is not synchronized, if L307 of committer.go is executed and a report is created before L308 is executed, an incorrect report may be generated.

Reply:
Modified.
I added `internal/storagenode/logstream.(reportCommitBase).invalid` rather than keeping two confused variables - `internal/storagenode/logstream.(reportCommitBase).llsnOffset` and `internal/storagenode/logstream.(logStreamContext).uncommittedLLSNBegin`. It there is inconsistency between the commit context and the last log entry, the `internal/storagenode/logstream.(reportCommitBase).invalid` will be set. Invalid reportCommitBase should be resolved by syncrhonization.